sealtk/core/test/KwiverPipelineWorker.cpp

```diff
@@ -0,0 +1,232 @@
+/* This file is part of SEAL-TK, and is distributed under the OSI-approved BSD
+ * 3-Clause License. See top-level LICENSE file or
+ * https://github.com/Kitware/seal-tk/blob/master/LICENSE for details. */
+
+#include <sealtk/test/TestCore.hpp>
+
+#include <sealtk/core/test/TestCommon.hpp>
+
+#include <sealtk/core/ImageUtils.hpp>
+#include <sealtk/core/KwiverPipelineWorker.hpp>
+#include <sealtk/core/KwiverVideoSource.hpp>
+
+#include <sealtk/util/unique.hpp>
+
+#include <sprokit/processes/adapters/adapter_data_set.h>
+
+#include <vital/algo/video_input.h>
+#include <vital/config/config_block.h>
+
+#include <vital/range/iota.h>
+
+// #include <arrows/qt/image_container.h>
+//
+#include <qtGet.h>
+#include <qtStlUtil.h>
+
+#include <QImage>
+
+#include <QtTest>
+
+namespace ka = kwiver::adapter;
+namespace kv = kwiver::vital;
+namespace kvr = kwiver::vital::range;
+
+namespace sealtk
+{
+
+namespace core
+{
+
+namespace test
+{
+
+// ============================================================================
+class TestPipelineWorker : public KwiverPipelineWorker
+{
+public:
+  TestPipelineWorker() : KwiverPipelineWorker{RequiresInputAndOutput} {}
+
+  QList<QHash<int, QString>> outputNames;
+  QList<QHash<int, QImage>> outputImages;
+
+protected:
+  void processOutput(ka::adapter_data_set_t const& output) override;
```

Comment:
This makes me wonder if there should instead be a signal emitted by `KwiverPipeilneWorker` that is invoked whenever output is received.

Reply:
Following up on this much later... I think it depends on which thread (and which class) is expected to turn the "raw" input (i.e. the `adapter_data_set_t`) into something that is meaningful to the GUI. In another branch (not yet PR'd), I wound up subclassing `KwiverPipelineWorker` further and overriding this method.